### ISSUE TITLE
axios.create() can't not handle params

### ISSUE THREAD

**Issue by Zu3zz** [axios/axios#2230]
**Describe the bug**
I created a instance with a param in it, but it doesn't seem to work

I Have already decleared a instance axios in my js file

```javascript
const instance = axios.create({
  baseURL: '/',
  params: {
    secret: 123
  }
})
```

I want to send a request like this
**/api/logout.json?secret=123**

```javascript
instance.get('/api/logout.json)
   .then((res) => {
   dispatch(changeLogin(false))
})
```

but after I use this instance to send a request
the request is only **/api/logout.json**

*What is this problem, did instance doesn't received params?*

**Comment by evenfrost** [comment]
It was introduced by `0.18`, `0.17` works well.

**Comment by waiaan** [comment]
```js
var instance = axios.create({
  //you should not add 'method' option here
});
```

**Comment by waiaan** [comment]
```js
var instance = axios.create({
  //you should not add 'method' option here
});
```

**Comment by zazapeta** [comment]
> ```js
> var instance = axios.create({
>   //you should not add 'method' option here
> });
> ```

Hi, 
Why we should not ? 

Thx,

zazapeta

**Comment by neuroid2018** [comment]
> **Describe the bug**
> I created a instance with a param in it, but it doesn't seem to work
> 
> I Have already decleared a instance axios in my js file
> 
> ```js
> const instance = axios.create({
>   baseURL: '/',
>   params: {
>     secret: 123
>   }
> })
> ```
> 
> I want to send a request like this
> **/api/logout.json?secret=123**
> 
> ```js
> instance.get('/api/logout.json)
>    .then((res) => {
>    dispatch(changeLogin(false))
> })
> ```
> 
> but after I use this instance to send a request
> the request is only **/api/logout.json**
> 
> _What is this problem, did instance doesn't received params?_

As a workaround create a config object instead of axios instance:
const config = {
   baseURL: '/',
   params: {
    secret: 123
  }
}
Then to send a GET request:
axios.request(config)

and axios should be imported:
import axios from 'axios'

**Comment by Zu3zz** [comment]
> > **Describe the bug**
> > I created a instance with a param in it, but it doesn't seem to work
> > I Have already decleared a instance axios in my js file
> > ```js
> > const instance = axios.create({
> >   baseURL: '/',
> >   params: {
> >     secret: 123
> >   }
> > })
> > ```
> > 
> > 
> > I want to send a request like this
> > **/api/logout.json?secret=123**
> > ```js
> > instance.get('/api/logout.json)
> >    .then((res) => {
> >    dispatch(changeLogin(false))
> > })
> > ```
> > 
> > 
> > but after I use this instance to send a request
> > the request is only **/api/logout.json**
> > _What is this problem, did instance doesn't received params?_
> 
> As a workaround create a config object instead of axios instance:
> const config = {
> baseURL: '/',
> params: {
> secret: 123
> }
> }
> Then to send a GET request:
> axios.request(config)
> 
> and axios should be imported:
> import axios from 'axios'

but there's someing else for me to use in axios instance, so if I want to add params, i shouldn't do it in instance but to add it in config?
Is that what u mean bro

**Comment by waiaan** [comment]
> @waiaan
> 
> > ```js
> > var instance = axios.create({
> >   //you should not add 'method' option here
> > });
> > ```
> 
> Hi,
> Why we should not ?
> 
> Thx,
> 
> zazapeta

it seems that some options would not be merged from instance.
from axios/lib/core/mergeConfig.js  /
```js
utils.forEach(['url', 'method', 'params', 'data'], function valueFromConfig2(prop) {
    if (typeof config2[prop] !== 'undefined') {
      config[prop] = config2[prop];
    }
  });
```

the 'config' is an emtpy object.

**Comment by waiaan** [comment]
> @waiaan
> 
> > ```js
> > var instance = axios.create({
> >   //you should not add 'method' option here
> > });
> > ```
> 
> Hi,
> Why we should not ?
> 
> Thx,
> 
> zazapeta

you may use 0.18.x , it would work.

**Comment by hamadr** [comment]
Hi,
I noticed that creating an object with
```
axios.create({
        baseURL: 'https://myapi.com',
        params: {
          my_param: 'a_value',
        },
      });
```
works in 0.18.x whereas it doesn't work anymore in 0.19.
I didn't see anything related to this in the release notes.
Should it be considered as a regression?

**Comment by waiaan** [comment]
> Hi,
> I noticed that creating an object with
> 
> ```
> axios.create({
>         baseURL: 'https://myapi.com',
>         params: {
>           my_param: 'a_value',
>         },
>       });
> ```
> 
> works in 0.18.x whereas it doesn't work anymore in 0.19.
> I didn't see anything related to this in the release notes.
> Should it be considered as a regression?

see what i said.

**Comment by neuroid2018** [comment]
> > > **Describe the bug**
> > > I created a instance with a param in it, but it doesn't seem to work
> > > I Have already decleared a instance axios in my js file
> > > ```js
> > > const instance = axios.create({
> > >   baseURL: '/',
> > >   params: {
> > >     secret: 123
> > >   }
> > > })
> > > ```
> > > 
> > > 
> > > I want to send a request like this
> > > **/api/logout.json?secret=123**
> > > ```js
> > > instance.get('/api/logout.json)
> > >    .then((res) => {
> > >    dispatch(changeLogin(false))
> > > })
> > > ```
> > > 
> > > 
> > > but after I use this instance to send a request
> > > the request is only **/api/logout.json**
> > > _What is this problem, did instance doesn't received params?_
> > 
> > 
> > As a workaround create a config object instead of axios instance:
> > const config = {
> > baseURL: '/',
> > params: {
> > secret: 123
> > }
> > }
> > Then to send a GET request:
> > axios.request(config)
> > and axios should be imported:
> > import axios from 'axios'
> 
> but there's someing else for me to use in axios instance, so if I want to add params, i shouldn't do it in instance but to add it in config?
> Is that what u mean bro

Yes. Consider it as a workaround until this regression will be fixed. 
If you want to use a predefined config but need to modify/add some params you have to either modify that config or create a new one.

**Comment by Zu3zz** [comment]
> > > > **Describe the bug**
> > > > I created a instance with a param in it, but it doesn't seem to work
> > > > I Have already decleared a instance axios in my js file
> > > > ```js
> > > > const instance = axios.create({
> > > >   baseURL: '/',
> > > >   params: {
> > > >     secret: 123
> > > >   }
> > > > })
> > > > ```
> > > > 
> > > > 
> > > > I want to send a request like this
> > > > **/api/logout.json?secret=123**
> > > > ```js
> > > > instance.get('/api/logout.json)
> > > >    .then((res) => {
> > > >    dispatch(changeLogin(false))
> > > > })
> > > > ```
> > > > 
> > > > 
> > > > but after I use this instance to send a request
> > > > the request is only **/api/logout.json**
> > > > _What is this problem, did instance doesn't received params?_
> > > 
> > > 
> > > As a workaround create a config object instead of axios instance:
> > > const config = {
> > > baseURL: '/',
> > > params: {
> > > secret: 123
> > > }
> > > }
> > > Then to send a GET request:
> > > axios.request(config)
> > > and axios should be imported:
> > > import axios from 'axios'
> > 
> > 
> > but there's someing else for me to use in axios instance, so if I want to add params, i shouldn't do it in instance but to add it in config?
> > Is that what u mean bro
> 
> Yes. Consider it as a workaround until this regression will be fixed.
> If you want to use a predefined config but need to modify/add some params you have to either modify that config or create a new one.

THX a lot, I will stop using params in instance

**Comment by evenfrost** [comment]
@Zu3zz 	I believe this should not be closed until a PR with the fix is created and merged to avoid duplicate issues. As been already said, this is a regression bug, and you don't have to stop using params in instance, rather don't use `0.19` until that. :)

**Comment by evenfrost** [comment]
Actually this is already a duplicate and is addressed in #2190 and #2196, so cancel that.
